en/lessons/ecto/changesets.md

```diff
@@ -0,0 +1,372 @@
+
+```elixir
+defmodule Example.Person do
+  use Ecto.Schema
+
+  schema "people" do
+    field :name, :string
+    field :age, :integer, default: 0
+  end
+end
+```
+
+### `cast/4`
+
+To create a changeset using the `Example.Person` schema, we are going to use [`Ecto.Changeset.cast/4`](https://hexdocs.pm/ecto/Ecto.Changeset.html#cast/4).
+The first parameter should be our starting data which for us will be an empty `%Person{}` but could be an existing record.
+The secord parameter `cast/4` expects is the data changes as a map.
+Our third parameter is a list of fields we're accepting values for, fields not included in this list will be ignored.
+The foruth and final parameter are changeset options which we'll explore later, for now we'll focus on the first three:
+
+```elixir
+iex> alias Example.Person
+iex> Ecto.Changeset.cast(%Person{}, %{"name" => "Jack"}, [:age, :name])
+#Ecto.Changeset<
+ action: nil,
+ changes: %{name: "Jack"},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+ ```
+
+We can see that we've created a new changeset and this time it's populated with values, most importantly: `:changes` and `:valid?`.
+
+Before we go too much further let's quickly take a look at how permitted fields work:
+
+ ```elixir
+iex> Ecto.Changeset.cast(%Person{}, %{"email" => "jack@example.com", "name" => "Jack"}, [:name])
+#Ecto.Changeset<
+ action: nil,
+ changes: %{name: "Jack"},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+```
+
+What we see is the same valid changeset from before with no mention of the `"email"` value because it was not explicitly permitted.
+
+### `change/2`
+
+If the source of the data change is truthworthy `change/2` can be a viable alternative to `cast/4`.
+The primary difference between these two functions is that `change/2` does no filtering or casting of values.
+
+To quickly demonstrate this let's assume a person's birthday is entered and age is computed from that.
+Since we perform this calculation ourelves we can trust that the value will be correctly formatted and cast:
+
+```elixir
+%Person{}
+|> Ecto.Changeset.cast(%Person{}, %{"name" => "Jack"}, [:name])
+|> Ecto.Changeset.change(%{age: 30})
+
+#Ecto.Changeset<
+ action: nil,
+ changes: %{age: 30, name: "Jack"},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+```
+
+## Validations
+
+We've learned how to create changesets and apply changes to them but how do we validate those changes?
+
+If we take our previous examples which lack validation, any changes to a person's name is considered valid which means a blank string counts:
+
+```elixir
+iex> Ecto.Changeset.cast(%Person{}, %{"name" => ""}, [:name])
+#Ecto.Changeset<
+ action: nil,
+ changes: %{name: ""},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+```
+
+This won't work for a real application so let's look at how we can handle this with Ecto.
+Lucky for us, as with many of the Elixir libraries, Ecto comes with a slew of awesome features one of which is built-in validaitons.
```

Comment:
Maybe change the wording for `Lucky for us` to something else? Like `Luckily for us, `, or `Thankfully`, or something else.

Reply:
@svileng please call these out!  I _try_ not to use a lot informal local terms but sometimes I don't catch myself.  One thing I look for in these reviews is things like this!  I want to make translation as easy as we can 😁